test/user_defined_output/log_response_headers_plugin_test.cc

```diff
@@ -0,0 +1,228 @@
+      "nighthawk.log_response_headers_plugin");
+  Envoy::ProtobufTypes::MessagePtr empty_config = factory.createEmptyConfigProto();
+  LogResponseHeadersConfig expected_config;
+  EXPECT_THAT(*empty_config, EqualsProto(expected_config));
+}
+
+TEST(LogResponseHeadersPluginFactory, FactoryRegistersUnderCorrectName) {
+  LogResponseHeadersConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& factory = Envoy::Config::Utility::getAndCheckFactoryByName<UserDefinedOutputPluginFactory>(
+      "nighthawk.log_response_headers_plugin");
+  EXPECT_EQ(factory.name(), "nighthawk.log_response_headers_plugin");
+}
+
+TEST(LogResponseHeadersPluginFactory, CreateUserDefinedOutputPluginCreatesCorrectPluginType) {
+  LogResponseHeadersConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& factory = Envoy::Config::Utility::getAndCheckFactoryByName<UserDefinedOutputPluginFactory>(
+      "nighthawk.log_response_headers_plugin");
+  UserDefinedOutputPluginPtr plugin = factory.createUserDefinedOutputPlugin(config_any, {});
+  EXPECT_NE(dynamic_cast<LogResponseHeadersPlugin*>(plugin.get()), nullptr);
+}
+
+TEST(GetPerWorkerOutput, ReturnsProtoOfCorrectType) {
+  std::unique_ptr<HeaderLogger> logger = nullptr;
+  UserDefinedOutputPluginPtr plugin = CreatePlugin("", std::move(logger));
+
+  absl::StatusOr<Envoy::ProtobufWkt::Any> any_or = plugin->getPerWorkerOutput();
+  EXPECT_TRUE(any_or.status().ok());
+  EXPECT_TRUE(any_or->Is<LogResponseHeadersOutput>());
+}
+
+TEST(HandleResponseHeaders, LogsAllHeadersIfConfigured) {
+  std::unique_ptr<FakeHeaderLogger> logger = std::make_unique<FakeHeaderLogger>();
+  FakeHeaderLogger* logger_ptr = logger.get();
+  UserDefinedOutputPluginPtr plugin =
+      CreatePlugin("logging_mode:LM_LOG_ALL_RESPONSES", std::move(logger));
+  TestResponseHeaderMapImpl headers{
+      {":status", "200"}, {"mytestheader1", "myvalue1"}, {"mytestheader2", "myvalue2"}};
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers).ok());
+  std::vector<const HeaderEntry*> logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_EQ(logged_headers.size(), 3);
+  EXPECT_EQ(logged_headers[0]->key().getStringView(), ":status");
+  EXPECT_EQ(logged_headers[0]->value().getStringView(), "200");
+  EXPECT_EQ(logged_headers[1]->key().getStringView(), "mytestheader1");
+  EXPECT_EQ(logged_headers[1]->value().getStringView(), "myvalue1");
+  EXPECT_EQ(logged_headers[2]->key().getStringView(), "mytestheader2");
+  EXPECT_EQ(logged_headers[2]->value().getStringView(), "myvalue2");
+}
+
+TEST(HandleResponseHeaders, LogsSpecifiedHeaders) {
+  std::unique_ptr<FakeHeaderLogger> logger = std::make_unique<FakeHeaderLogger>();
+  FakeHeaderLogger* logger_ptr = logger.get();
+  UserDefinedOutputPluginPtr plugin = CreatePlugin(R"(logging_mode:LM_LOG_ALL_RESPONSES
+      log_headers_with_name: "mytestheader1"
+      log_headers_with_name: "mytestheader2"
+  )",
```

Comment:
Is this the outcome of autoformatting?

Reply:
Sort of. I have control over lines 141-143, but not over where line 144 is. I think I improved it.

Reply:
👍